pkg/kubelet/gpu/nvidia/nvidia_gpu_manager.go

```diff
@@ -101,8 +102,7 @@ func (ngm *nvidiaGPUManager) Start() error {
 	if err := ngm.discoverGPUs(); err != nil {
 		return err
 	}
-	// It's possible that the runtime isn't available now.
-	ngm.allocated = ngm.gpusInUse()
+
```

Comment:
I tried to figure out the right place to call Start() function, but seems pods' synchronization is done by the statusManager concurrently, which means there is no explicit checkpoint to tell if the pods have been recovered. So, if we can not get the right info of active pods in gpusInUse(), the initialization of ngm.allocated will never be right when kubelet restarts.